src/pages/Jam.svelte

```diff
@@ -38,13 +38,15 @@
     const historyLength = 2;
 
     let audioDevice: MediaDeviceInfo;
-    $: handleDeviceSelect(audioDevice);
-    function handleDeviceSelect(device?: MediaDeviceInfo) {
+
+    // FIXME - too many async/await here
+    $: async () => await handleDeviceSelect(audioDevice);
```

Comment:
What is this doing?

Can you just return the Promise?
```suggestion
    $:  () => handleDeviceSelect(audioDevice);
```